stage0/run.go

```diff
@@ -188,6 +188,24 @@ func MergeMounts(mounts []schema.Mount, appMounts []schema.Mount) []schema.Mount
 	return deduplicateMPs(ml)
 }
 
+// MergePorts merges two port lists
+func MergePorts(a, b []types.Port) []types.Port {
```

Comment:
I'm not sure how relevant this maybe in this context, but this is not a stable merge (i.e. it doesn't guarantee too keep the relative order). It may poses some issues when trying to answer questions like `what is the precedence order between manifest ports and exposed ports`.